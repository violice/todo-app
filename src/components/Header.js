import React from 'react';
import { createUseStyles } from 'react-jss'

const useStyles = createUseStyles({
  headerContainer: {
    backgroundColor: '#673ab7',
    boxShadow: '0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12)',
    color: '#fff',
    marginBottom: '64px',
  },
  headerContent: {
    maxWidth: '1200px',
    height: '64px',
    margin: '0 auto',
    padding: '0px 24px',
    display: 'flex',
    alignItems: 'center',
  },
  logo: {
    fontSize: '24px',
    fontWeight: 500,
  },
});

const Header = () => {
  const styles = useStyles();
  return (
    <header className={styles.headerContainer}>
      <div className={styles.headerContent}>
        <span className={styles.logo}>TODO</span>
      </div>
    </header>
  )
};

export default Header;