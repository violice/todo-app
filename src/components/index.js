export { default as AuthForm } from './AuthForm';
export { default as TaskForm } from './TaskForm';
export { default as Header } from './Header';
export { default as TasksList } from './TasksList';