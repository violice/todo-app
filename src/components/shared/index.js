export { default as Button } from './Button';
export { default as LoadingIndicator } from './LoadingIndicator';
export { default as Message } from './Message';
export { default as TextField } from './TextField';