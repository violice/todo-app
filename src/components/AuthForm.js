import React from 'react';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';
import { Button, TextField, Message, LoadingIndicator } from 'components/shared';

const useStyles = createUseStyles({
  container: {
    maxWidth: '320px',
    padding: '32px',
    margin: '0 auto',
    backgroundColor: '#fff',
    boxShadow: '0 2px 1px -1px rgba(0,0,0,.2), 0 1px 1px 0 rgba(0,0,0,.14), 0 1px 3px 0 rgba(0,0,0,.12)',
    borderRadius: '8px',
    position: 'relative',
  },
  messageContainer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  buttonContainer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    marginTop: '32px',
  },
});

const AuthForm = ({ username, password, isLoading, error, onUsernameChange, onPasswordChange, onFormSubmit }) => {
  const styles = useStyles();

  const onSubmit = (e) => {
    e.preventDefault();
    onFormSubmit({ username, password });
  }

  return (
    <form className={styles.container} onSubmit={onSubmit}>
      {isLoading && <LoadingIndicator withOverlay />}
      <TextField title="Username" value={username} onChange={onUsernameChange} />
      <TextField title="Password" value={password} onChange={onPasswordChange} />
      {error &&
        <div className={styles.messageContainer}>
          <Message value={error.message} />
        </div>
      }
      <div className={styles.buttonContainer}>
        <Button title="SignIn / SignUp" />
      </div>
    </form>
  );
};

AuthForm.propTypes = {
  username: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  isLoading: PropTypes.bool,
  error: PropTypes.object,
  onUsernameChange: PropTypes.func.isRequired,
  onPasswordChange: PropTypes.func.isRequired,
  onFormSubmit: PropTypes.func.isRequired,
};

AuthForm.defaultProps = {
  isLoading: false,
  error: null,
};

export default AuthForm;