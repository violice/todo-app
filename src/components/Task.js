import React from 'react';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';
import { TaskForm } from 'components';
import { LoadingIndicator } from './shared';

const useStyles = createUseStyles({
  container: {
    maxWidth: '320px',
    padding: '32px',
    margin: '0 auto',
    backgroundColor: '#fff',
    boxShadow: '0 2px 1px -1px rgba(0,0,0,.2), 0 1px 1px 0 rgba(0,0,0,.14), 0 1px 3px 0 rgba(0,0,0,.12)',
    borderRadius: '8px',
    marginBottom: '16px',
    display: 'flex',
    flexDirection: 'row',
    position: 'relative',
  },
  leftPart: {
    flex: '1',
  },
  rightPart: {
    display: 'flex',
    alignItems: 'center',
    '&>i': {
      '&:hover': {
        cursor: 'pointer',
      },
      '&:last-child': {
        color: '#FF1A43',
      },
    },
  },
  title: {
    fontSize: '16px',
    fontWeight: '500',
  },
  body: {
    fontSize: '14px',
  },
});

const Task = ({
  task,
  editedTask,
  deletedTask,
  isTaskEditing,
  isTaskDeleting,
  taskEditingError,
  onEditedTaskChange,
  onDeletedTaskChange,
  onTaskEdit,
  onTaskDelete,
}) => {
  const styles = useStyles(); 

  if (editedTask.id === task.id && editedTask.mode === 'edit') {
    return (
      <TaskForm
        task={editedTask}
        isLoading={isTaskEditing}
        error={taskEditingError}
        onTaskChange={onEditedTaskChange}
        onFormSubmit={onTaskEdit}
      />
    );
  }

  return (
    <div className={styles.container}>
      {(((editedTask.id === task.id) && isTaskEditing) || (isTaskDeleting && (task.id === deletedTask.id))) && <LoadingIndicator withOverlay />}
      <div className={styles.leftPart}>
        <div className={styles.title}>{task.title}</div>
        <div className={styles.body}>{task.body}</div>
      </div>
      <div className={styles.rightPart}>
        {task.completed
          ? <i className="material-icons" onClick={() => {
            onEditedTaskChange({ ...task, mode: 'default' });
            onTaskEdit({ ...task, completed: false });
          }}>check_box</i>
          : <i className="material-icons" onClick={() => {
            onEditedTaskChange({ ...task, mode: 'default' });
            onTaskEdit({ ...task, completed: true });
          }}>check_box_outline_blank</i>
        }
        <i className="material-icons" onClick={() => onEditedTaskChange({ ...task, mode: 'edit' })}>edit</i>
        <i className="material-icons" onClick={() => {
          onDeletedTaskChange(task);
          onTaskDelete(task.id);
        }}>close</i>
      </div>
    </div >
  );
};

Task.propTypes = {
  task: PropTypes.object.isRequired,
  editedTask: PropTypes.object.isRequired,
  deletedTask: PropTypes.object.isRequired,
  isTaskEditing: PropTypes.bool.isRequired,
  isTaskDeleting: PropTypes.bool.isRequired,
  taskEditingError: PropTypes.object,
  onEditedTaskChange: PropTypes.func.isRequired,
  onDeletedTaskChange: PropTypes.func.isRequired,
  onTaskEdit: PropTypes.func.isRequired,
  onTaskDelete: PropTypes.func.isRequired,
};

Task.defaultProps = {
  taskEditingError: null,
};

export default Task;