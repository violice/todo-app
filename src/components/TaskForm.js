import React from 'react';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';
import { Button, TextField, Message, LoadingIndicator } from 'components/shared';

const useStyles = createUseStyles({
  container: {
    maxWidth: '320px',
    padding: '32px',
    margin: '0 auto',
    backgroundColor: '#fff',
    boxShadow: '0 2px 1px -1px rgba(0,0,0,.2), 0 1px 1px 0 rgba(0,0,0,.14), 0 1px 3px 0 rgba(0,0,0,.12)',
    borderRadius: '8px',
    marginBottom: '16px',
    position: 'relative',
  },
  messageContainer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  buttonContainer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    marginTop: '32px',
  },
});

const TaskForm = ({ task, isLoading, error, onTaskChange, onFormSubmit }) => {
  const styles = useStyles();

  const onSubmit = (e) => {
    e.preventDefault();
    onFormSubmit(task);
  };

  return (
    <form className={styles.container} onSubmit={onSubmit}>
      {isLoading && <LoadingIndicator withOverlay />}
      <TextField title="Title" value={task.title} onChange={newTitle => onTaskChange({ ...task, title: newTitle })} />
      <TextField title="Body" value={task.body} multiline onChange={newBody => onTaskChange({ ...task, body: newBody })} />
      {error &&
        <div className={styles.messageContainer}>
          <Message value={error.message} />
        </div>
      }
      <div className={styles.buttonContainer}>
        <Button title={task.id ? 'Edit task' : 'Create task'} />
      </div>
    </form>
  );
};


TaskForm.propTypes = {
  task: PropTypes.object.isRequired,
  isLoading: PropTypes.bool,
  error: PropTypes.object,
  onTaskChange: PropTypes.func.isRequired,
  onFormSubmit: PropTypes.func.isRequired,
};

TaskForm.defaultProps = {
  isLoading: false,
  error: null,
};

export default TaskForm;