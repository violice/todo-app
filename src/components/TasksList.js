import React from 'react';
import PropTypes from 'prop-types';
import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';

import Task from './Task';

const TasksList = ({
  tasks,
  editedTask,
  deletedTask,
  isTaskEditing,
  isTaskDeleting,
  taskEditingError,
  onEditedTaskChange,
  onDeletedTaskChange,
  onTaskEdit,
  onTaskDelete,
  onTasksReordered,
}) => {
  return (
    <div>
      <DragDropContext onDragEnd={({ source, destination }) => {
        if (destination && source.index !== destination.index) {
          let newTasks = [...tasks];
          const [removed] = newTasks.splice(source.index, 1);
          newTasks.splice(destination.index, 0, removed);
          onTasksReordered(newTasks.map((task, index) => ({ ...task, order: index })));
        }
      }}>
        <Droppable droppableId="droppable">
          {(provided) => (
            <div
              {...provided.droppableProps}
              ref={provided.innerRef}
            >
              {
                tasks.map((task, index) => (
                  <Draggable key={task.id} draggableId={task.id} index={index}>
                    {(provided) => (
                      <div
                        ref={provided.innerRef}
                        {...provided.draggableProps}
                        {...provided.dragHandleProps}
                      >
                        <Task
                          task={task}
                          editedTask={editedTask}
                          deletedTask={deletedTask}
                          isTaskEditing={isTaskEditing}
                          isTaskDeleting={isTaskDeleting}
                          taskEditingError={taskEditingError}
                          onEditedTaskChange={onEditedTaskChange}
                          onDeletedTaskChange={onDeletedTaskChange}
                          onTaskEdit={onTaskEdit}
                          onTaskDelete={onTaskDelete}
                        />
                      </div>
                    )}
                  </Draggable>

                ))
              }
            </div>
          )}
        </Droppable>
      </DragDropContext>
    </div>
  )
};

TasksList.propTypes = {
  tasks: PropTypes.arrayOf(PropTypes.object).isRequired,
  editedTask: PropTypes.object.isRequired,
  deletedTask:  PropTypes.object.isRequired,
  isTaskEditing: PropTypes.bool.isRequired,
  isTaskDeleting: PropTypes.bool.isRequired,
  taskEditingError: PropTypes.object,
  onEditedTaskChange: PropTypes.func.isRequired,
  onDeletedTaskChange: PropTypes.func.isRequired,
  onTaskEdit: PropTypes.func.isRequired,
  onTaskDelete: PropTypes.func.isRequired,
  onTasksReordered: PropTypes.func.isRequired,
};

TasksList.defaultProps = {
  taskEditingError: null,
};

export default TasksList;