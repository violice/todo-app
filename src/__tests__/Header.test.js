import React from 'react';
import { render } from '@testing-library/react';
import { Header } from 'components';

it('should take a snapshot', () => {
  const { asFragment } = render(<Header />)
  expect(asFragment()).toMatchSnapshot();
});
