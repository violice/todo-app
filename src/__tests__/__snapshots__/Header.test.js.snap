// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <header
    class="headerContainer-0-2-1"
  >
    <div
      class="headerContent-0-2-2"
    >
      <span
        class="logo-0-2-3"
      >
        TODO
      </span>
    </div>
  </header>
</DocumentFragment>
`;
